subsys/net/ip/utils.c

```diff
@@ -679,10 +679,22 @@ static bool parse_ipv6(const char *str, size_t str_len,
 	}
 
 	if ((ptr + 1) < (str + str_len) && *(ptr + 1) == ':') {
-		len = str_len - end;
+		/* -1 as end does not contain first [
+		 * -2 as pointer is advanced by 2, skipping ]:
+		 */
+		len = str_len - end - 1 - 2;
+
+		ptr += 2;
+
+		for (i = 0; i < len; i++) {
```

Comment:
Why do we need to check if there is null char here? I mean that the caller has specified the max length already, and the `convert_port()` will return failure if the value is not a valid integer. So shouldn't we fail if the string len is wrong, instead of returning a truncated port number.

Reply:
This is based on the assumption the test cases are correct and one of them is testing the case where there is a mismatch between string length and the actual string:

https://github.com/zephyrproject-rtos/zephyr/blob/650d44384384ba531c34b21eb698841fb512fbc6/tests/net/utils/src/main.c#L721-L722

Reply:
If I am reading your patch correctly, isn't it now possible that if application calls
`net_ipaddr_parse("[2001:db8::3]:123\04", sizeof("[2001:db8::3]:123\04") - 1, &addr)`
we would get "123" as a port instead of error?

Reply:
I did not tested this but to me it looks that the code before this patch would also have returned `123` as the string is `[2001:db8::3]:123`  (`4` is not part of the string as it come after `\0` and `strtoul` only looks in a string)

What I was wondering when I made the patch is if the test case is correct, if the test case is correct there is actually no need for the parameter `str_len` as the parse function it self determines the length.

Reply:
> no need for the parameter `str_len`

The purpose of the str_len field was that you can have a non-null terminated string and parse that.